### PR TITLE
Add integration test suite

### PR DESCRIPTION
**Pull request by per-samuelsson**
Issued in #4.

Also fixes a few bugs found during testing.

### CODE REVIEW

File: test/Starcounter.Database.Extensions.IntegrationTests/DbStorage.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        }
+
+        public ulong Delete(object obj)
+        {
+            if (obj == null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
+
+            int id = _objects.FirstOrDefault(t => object.ReferenceEquals(t.Value, obj)).Key;
```
Comment (miyconst) [inline test/Starcounter.Database.Extensions.IntegrationTests/DbStorage.cs:34]: Could be replaced with:

```suggestion
            int id = _objects.First(t => object.ReferenceEquals(t.Value, obj)).Key;
```

Then the following `if (id == 0)` can be removed.

File: test/Starcounter.Database.Extensions.IntegrationTests/DbStorage.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        }
+
+        public ulong GetOid(object obj)
+        {
+            if (obj == null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
+
+            int id = _objects.FirstOrDefault(t => object.ReferenceEquals(t.Value, obj)).Key;
```
Comment (miyconst) [inline test/Starcounter.Database.Extensions.IntegrationTests/DbStorage.cs:62]: Same as above, can be replaced with:

```suggestion
            int id = _objects.First(t => object.ReferenceEquals(t.Value, obj)).Key;
```

**Review by miyconst — APPROVED** [review]

File: test/Starcounter.Database.Extensions.IntegrationTests/DbStorage.cs (hunk 1 above)
Comment (per-samuelsson) [reply test/Starcounter.Database.Extensions.IntegrationTests/DbStorage.cs:34]: No. That would raise a `InvalidOperationException`. I'm raising a `ArgumentOutOfRangeException`.

File: test/Starcounter.Database.Extensions.IntegrationTests/DbStorage.cs (hunk 2 above)
Comment (per-samuelsson) [reply test/Starcounter.Database.Extensions.IntegrationTests/DbStorage.cs:62]: See above.

File: test/Starcounter.Database.Extensions.IntegrationTests/DbStorage.cs (hunk 1 above)
Comment (miyconst) [reply test/Starcounter.Database.Extensions.IntegrationTests/DbStorage.cs:34]: Yes, it's a different exception.

**Comment by per-samuelsson** [comment]
@miyconst, FYI, I dropped the `NuGet.Config` before merging. I'd like to keep this repo as tidy as possible to start with. I would even like to drop the `.sln`, since I don't want us to be so MS-tooling-oriented, but let it be there still, for your sake. 😄

**Pull request merged**